test/bandit/http2/plug_test.exs

```diff
@@ -214,11 +214,10 @@ defmodule HTTP2PlugTest do
     response = Req.head!(context.req, url: "/header_write_test")
 
     assert response.status == 200
-    assert map_size(response.headers) == 5
+    assert map_size(response.headers) == 4
 
     assert %{
              "date" => [date],
-             "content-length" => ["0"],
```

Comment:
This evidences a "breaking change" for existing users who depend on HEAD always sending back "content-length: 0".

As mentioned in an earlier comment, the existing behavior is arguably not correct, but [it has been enshrined](https://xkcd.com/1172/).

Reply:
I have [added tests](https://github.com/mtrudel/bandit/pull/353/commits/7018508b549a5ce4f3eb742e257a4577e9d844a0) to exercise how an application would explicitly set "content-length: 0" on HEAD responses:

```elixir
conn
|> put_resp_header("content-length", "0")
|> send_resp(200, "")
```

Unfortunately, I can't see a feasible way to disambiguate "incidentally zero-length" (i.e. `send_resp(200, "")`) from "intentionally zero-length" (i.e. `send_resp(200, generated_content) when byte_size(generated_content) == 0`) in the computed response body size.

Reply:
Yep this is a fine breaking change - for one it's super subtle, and for two, it's not something anyone would get themselves except deliberately. Thanks for surfacing it though!